data/tutorials/language/0it_06_imperative.md

```diff
@@ -331,6 +331,107 @@ In this example, the first two expressions are `print_endline` function calls, w
 
 Here, the semicolon after 42 is ignored.
 
+**`begin … end` expressions**
+
+In OCaml, `begin … end` and parentheses are the same.
+
+Imagine we want to write a function that:
+1. Has an `int` reference parameter containing value _n_
+2. Updates the reference's contents to _2 &times; (n + 1)_
+
+This is arguably convoluted and does not work:
+```ocaml
+# let f r = r := incr r; 2 * !r;;
+Error: This expression has type unit but an expression was expected of type int
+```
+
+But here is how it can be made to work:
+```ocaml
+# let f r = r := begin incr r; 2 * !r end;;
+val f : int ref -> unit = <fun>
+```
+
+The error came from assign `:=` associating stronger that semicolon `;`. Here is what we want to do, in order:
+1. Increment `r`
+2. Compute `2 * !r`
+3. Assign into `r`
+
+Remember the value of a semicolon-separated sequence is the value of its last expression. Grouping the first two steps with `begin … end` fixes the error.
+
+**Fun fact**: `begin … end` and parentheses are literally the same:
```

Comment:
This was said on li 336. If you meant to repeat it here, then that's fine, although you might want to rephrase it to something like " **Remember: .... **

Reply:
This is repeated here because the `unit` value, usually written `()` can also be written `begin end`. Useless by funny